kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -375,13 +382,12 @@ void onConnectionFailed(HostPort remote, Future<?> future, Channel inboundChanne
                 });
     }
 
-    private void onConnection(List<FilterAndInvoker> filters) {
+    private void onConnection() {
         state = State.CONNECTED;
         LOGGER.trace("{}: Outbound connected", inboundCtx.channel().id());
-        // Now we know which filters are to be used we need to update the DecodePredicate
-        // so that the decoder starts decoding the messages that the filters want to intercept
-        dp.setDelegate(DecodePredicate.forFilters(filters));
-        outboundChannelUsable();
+        if (outboundCtx != null) {
```

Comment:
Arguably this implies that we are missing a state for TLS negotiation in the connection state machine. As there is effectively a race between the future completing and the `outboundConnectionActive` firing. 

This future completes prior to `outboundConnectionActive` in the TCP case and after in the TLS case. 

cc: @tombentley